### PR TITLE
fix(outcomes): Do not report metrics-based outcomes for profiles

### PR DESCRIPTION
**Pull request by jjbayer**
Profiles are no longer dropped by dynamic sampling, so we do not need to track the virtual "total profile" category with a tag on metrics.

This also prevents double-reporting (the rate limiter on the payload also emits an outcome for the `profile` category).

ref: https://github.com/getsentry/sentry/pull/81799

### CODE REVIEW

File: relay-server/src/metrics/minimal.rs [hunk 1]
```diff
@@ -42,9 +40,7 @@ impl TrackableBucket for MinimalTrackableBucket {
                     MinimalValue::Counter(c) if mri.name == "usage" => c.to_f64() as usize,
                     _ => 0,
                 };
-                let has_profile = matches!(mri.name.as_ref(), "usage" | "duration")
-                    && self.tags.has_profile.is_some();
-                BucketSummary::Transactions { count, has_profile }
+                BucketSummary::Transactions(count)
```
Comment (Dav1dde) [inline relay-server/src/metrics/minimal.rs:43]: Can use the same formatting as for spans here:

```rs
            MetricNamespace::Transactions => BucketSummary::Transactions(match self.value {
                MinimalValue::Counter(c) if mri.name == "usage" => c.to_f64() as usize,
                _ => 0,
            }),
```

**Review by iambriccardo — APPROVED** [review]

**Review by Dav1dde — APPROVED** [review]
Nice!

Should double check if we can find any other mention of `has_profile` in (get)-sentry. I can't except in this [test](https://github.com/getsentry/sentry/blob/0396c6e78c02da86b4b5d7ca398743966dd14e03/tests/sentry/ingest/billing_metrics_consumer/test_billing_metrics_consumer_kafka.py#L65) 👍 .

**Comment by jjbayer** [comment]
> Should double check if we can find any other mention of `has_profile` in (get)-sentry.

I found [this constant](https://github.com/getsentry/sentry/blob/0396c6e78c02da86b4b5d7ca398743966dd14e03/src/sentry/search/events/constants.py#L331), added in https://github.com/getsentry/sentry/pull/64203.

@wmak is that list based on alert definitions that people actually use?

**Comment by jjbayer** [comment]
Decided we can merge this because the `has_profile` tag is only on `c:transactions/usage@none`, which is only used in the metrics billing consumer. I.e the tag cannot be accessed through the product.

**Pull request merged**